src/FshCompletionProvider.ts

```diff
@@ -131,6 +138,199 @@ export class FshCompletionProvider implements CompletionItemProvider {
+      const sortedEntities = possibleEntities.sort((a, b) =>
+        b.info.location.range.start.compareTo(a.info.location.range.start)
+      );
+      const targetEntity = sortedEntities.find(entity => {
+        return entity.info.location.range.start.isBeforeOrEqual(position);
+      });
+      if (targetEntity != null) {
+        if (targetEntity.info.type === 'Profile' && targetEntity.info.parent != null) {
+          return {
+            baseDefinition: targetEntity.info.parent,
+            existingPath
+          };
+        } else if (targetEntity.info.type === 'Instance' && targetEntity.info.instanceOf != null) {
+          return {
+            baseDefinition: targetEntity.info.instanceOf,
+            existingPath
+          };
+        } else if (targetEntity.info.type === 'Extension') {
+          return {
+            baseDefinition: targetEntity.info.parent ?? 'Extension',
+            existingPath
+          };
+        }
+      }
+    }
+    return null;
+  }
+
+  public getBaseDefinitionElements(entityName: string): ElementInfo[] {
+    // is the entity something that is locally defined?
+    // if so, look up parents until we get something not locally defined,
+    // or until we detect a circular reference.
+    let entityToCheck = entityName;
+    const checkedEntities: string[] = [];
+    while (this.definitionProvider.nameInformation.has(entityToCheck)) {
+      if (checkedEntities.includes(entityToCheck)) {
+        // circular reference escape hatch
+        return null;
+      }
+      checkedEntities.push(entityToCheck);
+      const entityInfo = this.definitionProvider.nameInformation.get(entityToCheck);
+      // take the first potentially useful thing from entityInfo
+      const infoToUse = entityInfo.find(info =>
+        ['Profile', 'Extension', 'Resource', 'Logical', 'Instance'].includes(info.type)
+      );
+      if (infoToUse == null) {
+        // the entity may be an alias that we are able to resolve
+        const aliasToUse = entityInfo.find(info => info.type === 'Alias');
+        if (aliasToUse) {
+          entityToCheck = aliasToUse.aliasValue;
+        } else {
+          // otherwise, we don't know
+          return null;
+        }
+      } else {
+        entityToCheck = infoToUse.parent ?? infoToUse.instanceOf;
+        // special handling for default parents:
+        // an Extension's default parent is Extension
+        // a Resource's default parent is DomainResource
+        // a Logical's default parent is Base
+        if (entityToCheck == null) {
+          if (infoToUse.type === 'Extension') {
+            entityToCheck = 'Extension';
+          } else if (infoToUse.type === 'Resource') {
+            entityToCheck = 'DomainResource';
+          } else if (infoToUse.type === 'Logical') {
+            entityToCheck = 'Base';
+          }
+        }
+      }
+    }
+    // is the entity present in the dependencies' entities?
+    // we may have to search twice, if we find a Profile
```

Comment:
When loading definitions in dependencies, elements don't get stored for profiles, since they should be the same as the elements we have for the profile's `baseDefinition`. So, if we find a profile, we do a second search using the `baseDefintion`. I'll clarify this in the comment.